### PR TITLE
clarify how a constraint is read

### PR DESCRIPTION
**Pull request by dontcallmedom**
* split up big block of text
* add "ReadConstraint" algorithm
* distinguish behavior in normal constraint vs advanced (cf https://github.com/w3c/mediacapture-main/issues/128)
* remove duplicate note about unknown properties

### CODE REVIEW

File: getusermedia.html [hunk 1]
```diff
@@ -3386,31 +3386,69 @@
...
+      <p>The behavior is as follows:</p>
+      <ol>
+        <li>all <var>min</var>, <var>max</var>, and <var>exact</var>
+      values in a constraint are together treated as
+      the <em>required</em> set</li>
+        <li>if it is not possible to satisfy simultaneously
+          all of those individual constraints for the indicated property names,
+          the User Agent MUST reject the returned promise.</li>
+        <li>Otherwise, it must
+      apply the required constraints.
```
Comment (alvestrand) [inline getusermedia.html]: I think this bullet should be removed. Applying the constraints comes much later.

Comment (dontcallmedom) [inline getusermedia.html]: Note that this text comes from the original text as is (the bullets just make it more visible)

**Comment by dontcallmedom** [comment]
I think @jan-ivar is right that this needs more work, and it's not clear this pull request is necessarily the best basis for that. I unfortunately won't have cycles to work on this myself for a while.

**Comment by alvestrand** [comment]
Linking to issue #128 which this fixes.

**Comment by alvestrand** [comment]
Closing, this will be addressed elsewhere.

**Pull request closed without merging**